src/axelrod_dojo/archetypes/gambler.py

```diff
@@ -55,6 +66,26 @@ def create_vector_bounds(self):
 
         return lb, ub
 
+    @staticmethod
+    def mutate_pattern(pattern, mutation_probability):
+        for i, _ in enumerate(pattern):
+            if random.random() <= mutation_probability:
+                pattern[i] = random.random()
```

Comment:
This is ok but it would work better if the mutation makes a minor adjustment rather than completely overwriting the value. In other words we're not randomly jumping around the space, rather we're exploring under the assumption that parameters close by each other typically have a similar fitness ("Hill-climbing").

Specifically, I would take the existing value of `pattern[i]`, add or subtract a random proportion between say -10% and 10%, and then normalize/truncate the value to [0,1] so that it's still a probability. The HMM player [does something similar](https://github.com/Axelrod-Python/axelrod-dojo/blob/master/src/axelrod_dojo/archetypes/hmm.py#L42).

For bonus points, you could write a function that could be used by both player types.

Reply:
> This is ok but it would work better if the mutation makes a minor adjustment rather than completely overwriting the value. In other words we're not trying to randomly jump around the space, rather we're exploring under the assumption that parameters close by each other typically have a similar fitness ("Hill-climbing").

I suspect this is one of the things where we could go back and forth for a while. I would suggest that for consistency with the status quo of the dojo that you go with @marcharper's suggestion @Nikoleta-v3.

At some point we could probably rigorously study the parametrization of the various heuristics and see what works best :+1:) I might do this with my final year project student next year.